akka-stream/src/main/scala/akka/stream/impl/FanoutProcessor.scala

```diff
@@ -79,11 +79,9 @@ private[akka] abstract class FanoutOutputs(val maxBufferSize: Int, val initialBu
     case SubscribePending ⇒
       subscribePending()
     case RequestMore(subscription, elements) ⇒
-      // FIXME can we avoid this cast?
       moreRequested(subscription.asInstanceOf[ActorSubscriptionWithCursor[Any]], elements)

```

Comment:
ah, now I remember, since it's in partial function we will not see if its another type, but I can add additional cases for that and throw something